File: src/main/java/net/ccbluex/liquidbounce/ui/client/gui/clickgui/style/styles/tenacity/impl/Component.java
/*
 * FDPClient Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge by LiquidBounce.
 * https://github.com/SkidderMC/FDPClient/
 */
package net.ccbluex.liquidbounce.ui.client.gui.clickgui.style.styles.tenacity.impl;

import net.minecraft.client.Minecraft;
public abstract class Component {

    public Minecraft mc = Minecraft.getMinecraft();

    abstract public void initGui();

    public abstract void keyTyped(char typedChar, int keyCode);

    abstract public void drawScreen(int mouseX, int mouseY);

    abstract public void mouseClicked(int mouseX, int mouseY, int button);

    abstract public void mouseReleased(int mouseX, int mouseY, int state);
}

File: src/main/java/net/ccbluex/liquidbounce/ui/client/clickgui/style/styles/fdpdropdown/impl/Component.kt
/*
 * FDPClient Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/SkidderMC/FDPClient/
 */
package net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.fdpdropdown.impl

abstract class Component {

    abstract fun initGui()

    abstract fun keyTyped(typedChar: Char, keyCode: Int)

    abstract fun drawScreen(mouseX: Int, mouseY: Int)

    abstract fun mouseClicked(mouseX: Int, mouseY: Int, button: Int)

    abstract fun mouseReleased(mouseX: Int, mouseY: Int, state: Int)

}